### PR TITLE
Tutorial: don't show more than one tutorial prompt at a time in the outfitter + show title

### PR DESCRIPTION
**Pull request by Hurleveur**
A big problem of this game is unknown features.
Part of those features are shown in the tutorial but because they overwhelm the player so much nobody reads them.

This just makes only one of them show per outfitter panel (tested, it works)

Also adds a small title of the tutorial:
OUTDATED (now shows `Multiple ships controls:`) as a title
Screenshot_2022-12-28_17-05-58

### CODE REVIEW

File: source/OutfitterPanel.cpp [hunk 1]
```diff
@@ -109,8 +109,8 @@ void OutfitterPanel::Step()
 	ShopPanel::Step();
 	ShopPanel::CheckForMissions(Mission::OUTFITTER);
 	if(GetUI()->IsTop(this) && !checkedHelp)
-		if(!DoHelp("outfitter") && !DoHelp("outfitter 2") && !DoHelp("outfitter 3"))
-			// All help messages have now been displayed.
+		if(DoHelp("outfitter") || DoHelp("outfitter 2") || DoHelp("outfitter 3") || true)
```
Comment (warp-core) [inline source/OutfitterPanel.cpp:112]: Is this:
`if(someExpression OR true)`
Do you want to know what the expression inside the if condition will evaluate to?
(Hint: it's always true.)

**Comment by warp-core** [comment]
Oh, I see, you're abusing short circuiting.

**Comment by warp-core** [comment]
If the issue is that people ignore the tutorial dialogs because they get them all at once, I'm not sure having a new one appear each of the first three times a person enters the outfitter is going to help.

Assumed current thought process:
*Enter outfitter*
*See tutorial*
*Read tutorial* (if people aren't even reading the first one, there's not much we can do for them.)
*Close dialog*
*New tutorial appears*
"This it too much reading."
*Skips tutorial*

What I imagine this person will do with your version:
*Enter outfitter*
*See tutorial*
*Read tutorial*
*Close dialog*
*Do outfitter things and leave*
*Assorted gameplay*
*Enter outfitter again*
*See tutorial*
"I already read the tutorial. Don't show me this again."
*Ignores tutorial*

**Comment by Hurleveur** [comment]
I don't think so, its obviously not the same tutorial? We don't show the tutorial multiple times do we?
~~also yes I love to abuse code uk~~

File: source/OutfitterPanel.cpp (hunk 1 above)
Comment (Hurleveur) [reply source/OutfitterPanel.cpp:112]: thats the point yes
unlimited powaaar ~~I could make it longer but its so elegant this way~~

**Comment by Hurleveur** [comment]
We could also make this advanced tutorial show up when it is relevant but that'd be a part of the new achivement etc class?

**Comment by Zitchas** [comment]
It really comes down to people's style. I personally love to read large text walls, but I still have a 50% chance that I just dismiss the second tutorial box in a row out of hand. That goes up to 75% for a third tutorial box. 

Speaking for myself, if that second and third tutorial get shifted to the second and third time I trigger something **and have something to let me easily identify that they are new and not repeats, like a title**, then I can virtually guarantee that I will read them all every time they appear.

**Comment by Hurleveur** [comment]
I'll add a title then.

**Comment by quyykk** [comment]
We really should have proper tutorial UI instead of using dialogs....

File: data/help.txt [hunk 2]
```diff
@@ -103,18 +103,18 @@ help "navigation"
 	`When you reach a new system, you can press <Land on planet / station> to land on any inhabited planets that are there.`
 	`Also, don't worry about crashing into asteroids or other ships; your ship will fly safely below or above them.`
 
-help "outfitter"
+help "outfitter basics"
```
Comment (quyykk) [inline data/help.txt:106]: Won't renaming those retrigger the tutorial for everybody?

Comment (warp-core) [reply data/help.txt:106]: Lots of people seem to have missed it, so...

Comment (Hurleveur) [reply data/help.txt:106]: we could keep this one named outfitter but many ppl def missed the other 2

**Comment by Hurleveur** [comment]
> We really should have proper tutorial UI instead of using dialogs....

totally agreed but in the meantime lets try to use dialogs a bit better

**Review by quyykk — APPROVED** [review]

File: data/help.txt (hunk 2 above)
Comment (Hurleveur) [reply data/help.txt]: is it ok if the 2 advanced tutorials trigger again? I think it is honestly

**Comment by 1010todd** [comment]
This might be outside the scope of this PR but maybe wider dialog boxes might help making these tutorials easier to read (or just use conversation panels.) Actually, if conversation panel is used maybe we could even make it more integrated into the game by having some character tell you the tutorials if possible(Though I can see how it might not work for in-flight tutorials if you declined the Intro.).

**Comment by Hurleveur** [comment]
Definitely outside the scope, and falls into what quyykk said of using special boxes for the tutos

File: source/OutfitterPanel.cpp [hunk 3]
```diff
@@ -109,8 +109,8 @@ void OutfitterPanel::Step()
 	ShopPanel::Step();
 	ShopPanel::CheckForMissions(Mission::OUTFITTER);
 	if(GetUI()->IsTop(this) && !checkedHelp)
-		if(!DoHelp("outfitter") && !DoHelp("outfitter 2") && !DoHelp("outfitter 3"))
-			// All help messages have now been displayed.
+		if(DoHelp("outfitter") || DoHelp("cargo management") || DoHelp("uninstalling and storage") || true)
```
Comment (Amazinite) [inline source/OutfitterPanel.cpp:112]: We definitely need a comment here to explain the use of short-circuiting.

File: source/Panel.cpp [hunk 4]
```diff
@@ -257,7 +257,7 @@ bool Panel::DoHelp(const string &name) const
 		return false;
 
 	Preferences::Set(preference);
-	ui->Push(new Dialog(message));
+	ui->Push(new Dialog(preference + "\n\n" + message));
```
Comment (Amazinite) [inline source/Panel.cpp:260]: If we're going to display titles, then they shouldn't be prefixed with `help: `, and should also be capitalized, so just giving the preference name isn't sufficient.

**Review by Amazinite — CHANGES_REQUESTED** [review]

File: source/OutfitterPanel.cpp [hunk 5]
```diff
@@ -109,8 +109,10 @@ void OutfitterPanel::Step()
 	ShopPanel::Step();
 	ShopPanel::CheckForMissions(Mission::OUTFITTER);
 	if(GetUI()->IsTop(this) && !checkedHelp)
-		if(!DoHelp("outfitter") && !DoHelp("outfitter 2") && !DoHelp("outfitter 3"))
-			// All help messages have now been displayed.
+		// Use short-circuiting to only display one of them at a time.
+		// (the first valid condition encountered will make us skip the others since it is an or)
```
Comment (Amazinite) [inline source/OutfitterPanel.cpp:113]:
```suggestion
		// Use short-circuiting to only display one of them at a time.
		// (The first valid condition encountered will make us skip the others.)
```

File: source/OutfitterPanel.cpp [hunk 6]
```diff
@@ -109,8 +109,10 @@ void OutfitterPanel::Step()
...
+		if(DoHelp("outfitter") || DoHelp("cargo management") || DoHelp("uninstalling and storage") || true)
+			// A help messages has now been displayed, or all of them.
```
Comment (Amazinite) [inline source/OutfitterPanel.cpp:115]:
```suggestion
			// Either a help message was freshly displayed, or all of them have already been seen.
```

**Review by Amazinite — APPROVED** [review]

**Pull request merged**